### ISSUE TITLE
Wrong propositional model when substitutions are enabled

### ISSUE THREAD

**Issue by blishko** [usi-verification-and-security/opensmt#123]
For the following propositional formula which is satisfiable, OpenSMT produces wrong model. The computed model is correct when substitutions are disabled.

```
(set-option :produce-model 1)
(set-logic QF_UF)
(declare-fun b () Bool)
(declare-fun c () Bool)
(assert (= b c ))
(assert (not b))
(check-sat)
(get-model)
``` 

Computed model is:
```
(define-fun b () Bool true)
(define-fun c () Bool true)
```

**Comment by blishko** [comment]
This is most likely also the cause of the issue #122

**Comment by blishko** [comment]
Fixed in #142
